components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,955 @@
+    hist_var%long_name = trim(longname)
+    hist_var%numdims   = length
+    hist_var%dtype     = dtype
+    ! Determine the dimension id's saved in the netCDF file and associated with
+    ! this variable
+    allocate(hist_var%dimid(length),hist_var%dimlen(length))
+    do dim_ii = 1,length
+      ierr = pio_inq_dimid(pio_atm_file%pioFileDesc,trim(dims(dim_ii)),hist_var%dimid(dim_ii))
+      call errorHandle("EAM_PIO ERROR: Unable to find dimension id for "//trim(dims(dim_ii)),ierr)
+      ierr = pio_inq_dimlen(pio_atm_file%pioFileDesc,hist_var%dimid(dim_ii),hist_var%dimlen(dim_ii))
+      call errorHandle("EAM_PIO ERROR: Unable to determine length for dimension "//trim(dims(dim_ii)),ierr)
+      if (hist_var%dimlen(dim_ii).eq.0) has_t_dim = .true.
+    end do
+
+    ! Distribute responsibility for writing cores over all PIO ranks
+    ! i.e. compute the degrees of freedom that this rank will contribute to PIO
+    if (has_t_dim) then
+      call get_compdof(length-1,hist_var%dimlen(:length-1),my_dof_len,istart,istop)
+    else
+      call get_compdof(length,hist_var%dimlen(:length),my_dof_len,istart,istop)
+    end if
+    allocate( hist_var%compdof(my_dof_len) )
+    hist_var%compdof(:my_dof_len) = (/ (ii, ii=istart,istop, 1) /)
+
+    ! Register Variable with PIO
+    ierr = PIO_def_var(pio_atm_file%pioFileDesc, trim(shortname), hist_var%dtype, hist_var%dimid(:length), hist_var%pioVar)
+    call errorHandle("PIO ERROR: could not define variable "//trim(shortname),ierr)
+
+    ! Gather the pio decomposition for this variable
+    if (has_t_dim) then
+      loc_len = max(1,length-1)
+      call get_decomp(tag,hist_var%dtype,hist_var%dimlen(:loc_len),hist_var%compdof,hist_var%iodesc)
+    else
+      call get_decomp(tag,hist_var%dtype,hist_var%dimlen,hist_var%compdof,hist_var%iodesc)
+    end if
+
+    return
+  end subroutine register_variable
+!=====================================================================!
+  subroutine get_pio_file(filename,pio_atm_file,found)
+
+    character(len=*), intent(in)   :: filename
+    type(pio_atm_output), pointer  :: pio_atm_file
+    logical, intent(out), optional :: found
+
+    type(pio_file_list), pointer :: curr => NULL(), prev => NULL()
+
+    ! Starting at the top of the current list of PIO_FILES find the next empty
+    ! one in list.
+    curr => pio_file_list_top
+    do while (associated(curr%pio_file))
+      if (trim(filename)==trim(curr%pio_file%filename)) then
+        pio_atm_file => curr%pio_file
+        if (present(found)) found = .true.
+        return
+      end if
+      prev => curr
+      curr => prev%next
+    end do
+    ! If we got this far then no pio_file with that name was found
+    if (present(found)) found = .false.
+    allocate(prev%next)
+    curr => prev%next
+    allocate(curr%pio_file)
+    pio_atm_file => curr%pio_file  
+
+  end subroutine get_pio_file
+!=====================================================================!
+  subroutine eam_h_write()
```

Comment:
h stands for header? Maybe better to change h=>header for clarity?

Reply:
also for eam_h_finalize, etc.

Reply:
This was borrowed from CAM too, I believe it stands for "history".  I made the change to be more clear.